src/app/data-planes/sources.ts

```diff
@@ -1,72 +1,89 @@
+import type { DataPlaneOverview as DataplaneOverview } from '@/types/index.d'
 import { normalizeFilterFields } from '@/utilities/normalizeFilterFields'
 
-type Closeable = { close: () => void }
+type CollectionParams = {
+  mesh: string
+}
+
+type DetailParams = CollectionParams & {
+  name: string
+}
+
 type PaginationParams = {
   size: number
   page: number
   search: string
 }
-type MeshParams = {
-  mesh: string
-}
+
 type ServiceParams = {
   service: string
 }
+
 type DataplaneTypeParams = {
   type: 'all' | 'delegated' | 'builtin'
 }
 
+type Closeable = { close: () => void }
+
+export type DataplaneOverviewSource = DataSourceResponse<DataplaneOverview>
 export type DataPlaneCollection = CollectionResponse<DataplaneOverview>
 export type DataPlaneCollectionSource = DataSourceResponse<DataPlaneCollection>
 
 export const sources = (api: KumaApi) => {
   return {
-    '/:mesh/dataplanes': async (params: MeshParams & PaginationParams, source: Closeable) => {
+    '/:mesh/dataplanes': async (params: CollectionParams & PaginationParams, source: Closeable) => {
       source.close()
+
+      const mesh = params.mesh
+      const size = params.size
       const offset = params.size * (params.page - 1)
-      return api.getAllDataplaneOverviewsFromMesh({
-        mesh: params.mesh,
-      }, {
-        ...Object.fromEntries(normalizeFilterFields(JSON.parse(params.search || '[]'))),
+      const gateway = 'false'
+      const filterParams = Object.fromEntries(normalizeFilterFields(JSON.parse(params.search || '[]')))
+
+      return api.getAllDataplaneOverviewsFromMesh({ mesh }, {
+        ...filterParams,
+        gateway,
         offset,
-        gateway: 'false',
-        size: params.size,
+        size,
       })
     },
-    '/:mesh/dataplanes/for/:service/of/:type': async (params: MeshParams & ServiceParams & PaginationParams & DataplaneTypeParams, source: Closeable) => {
+
+    '/:mesh/dataplane-overviews/:name': (params: DetailParams, source: Closeable) => {
       source.close()
+
+      const mesh = params.mesh
+      const name = params.name
+
+      return api.getDataplaneOverviewFromMesh({ mesh, name })
+    },
+
+    '/:mesh/dataplanes/for/:service/of/:type': async (params: CollectionParams & ServiceParams & PaginationParams & DataplaneTypeParams, source: Closeable) => {
+      source.close()
+
+      const mesh = params.mesh
+      const size = params.size
       const offset = params.size * (params.page - 1)
+
       // here 'all' means both proxies/sidecars and gateways this currently fits
       // our usecases but we should probably include `gateway | sidecar` or
       // similar
-      const search = Object.fromEntries(normalizeFilterFields(JSON.parse(params.search || '[]')))
-      if (typeof search.tag === 'undefined') {
-        search.tag = []
+      const filterParams = Object.fromEntries(normalizeFilterFields(JSON.parse(params.search || '[]')))
+      if (typeof filterParams.tag === 'undefined') {
```

Comment:
why is this a typeof? shouldn't this just be 

```
filterParams.tag === undefined
```
?

Reply:
I prefer the condition style to be `filterParams.tag === undefined` as well. As far as I know, it makes no difference. I could change this, but it’s just a difference in "programmer preference" that I don’t feel too strongly about, so I’m rather leaving it as is.

> filterParams.tag = filterParams.tag === undefined ? [] : filterParams.tag

As for this, I do prefer the style of using an if statement as I’m not unnecessarily assigning a property (even though here it would be assigning its original value).